snooty/test_parser.py

```diff
@@ -87,6 +87,24 @@ def test_tabs() -> None:
     assert isinstance(diagnostics[1], UnknownTabset)
     assert isinstance(diagnostics[2], DocUtilsParseError)
 
+    # Test a tab with no tabid
+    parser = rstparser.Parser(project_config, JSONVisitor)
+    page, diagnostics = parse_rst(
+        parser,
+        tabs_path,
+        """
+.. tabs-drivers::
+
+   .. tab::
+
+      Tab with no ID
+""",
+    )
+    page.finish(diagnostics)
+    assert [(type(d), d.severity) for d in diagnostics] == [
+        (DocUtilsParseError, Diagnostic.Level.error)
+    ]
+
```

Comment:
What's the expected output here? Is the `tab` directive omitted?

Reply:
Good question. I consider the exact output of any case where an error-level diagnostic raised to be implementation-defined. If it would make your life easier in some way, I'm happy to redefine the behavior as somewhere between "tabs directive is empty" and "this tab is excluded"

Reply:
Oops, sorry for overlooking this!

Both are equally easy to handle on the front end, but it feels preferable to me to omit each tab without `tabid`—let's go with that unless you have a different opinion. Thanks Andrew!

Reply:
New logic is a little gross, but it does the trick!